3.0.1-0 update with some fixes

jollaman999 opened the pull request.

bigbiff committed:
ADB: Add adb backup for TWRP.

Functionality for client side to backup
tar and image streams over adbd to the client under backup.ab.

Using adb backup on the client side you can backup the partitions
TWRP knows about.

On the client side you can do the following:
adb backup -f <filename> --twrp <options> where options are
--compress: compress data
system: backup system
cache: backup cache
data: backup data
boot: backup boot
etc for each partition.

You can string multiple options,
i.e. adb backup -f <filename> --twrp --compress cache system data

adb backup in TWRP will take any option corresponding
to TWRP fstab partitions, e.g. efs boot as well.

If you do not specify the filename with the -f option,
adb will backup your data to a filename backup.ab on the client.
You can then rename the file and encrypt it with desktop tools.

If you don't want to use command line arguments:
adb backup --twrp

will bring up the gui and allow you to choose partitions
from the backup page.

To restore the backup use the following convention:
adb restore <filename>

Structures are used to store metadata in binary inside
of the file itself. If the metadata structure is modified,
update the adb version so that it will invalidate older
backups and not cause issues on restore. When restoring,
we currently do not support picking specific partitions.
It's all or nothing.

Change-Id: Idb92c37fc9801dc8d89ed2a4570e9d12e76facf8

andi34 committed:
twrp: update Dutch translations

Change-Id: Ibcad999a1276b4519fc2b104b458ee77c984dc8b

Dees-Troy committed:
Fix no exfat-fuse rules

If we set TW_NO_EXFAT then we should not try to build exfat-fuse

Change-Id: I836ed6082cfbfb51bfb862b79c0182f36de88a9c

AdrianDC committed:
twrp: Add support for TW_SECONDARY_BRIGHTNESS_PATH

 * Required for devices like Sony Huashan (dual backlighting paths)

Change-Id: I0f84623431aec91fafee6617c1d4c542e4566caf

jcadduono committed:
Remove inexplicable checks for enabling/disabled UMS support

Removing these lines allows UMS to be enabled for devices with storage lun.
The checks don't seem to make sense as they are referencing always defined strings.

Change-Id: I75df9b7db5a790807fc2b5c946b271b103326b39

that1 committed:
minuitwrp: fix and hopefully speed up fbdev screen flipping

Fix: use row_bytes instead of xres (should help on Shield tablet)
Speed: Moving the calculations out of the inner loop

Change-Id: Ie43ae5e94ae88822360900c7b4d852b7aab4379b

Commit:
gui: add italian translation

Change-Id: Ia0bf0a3e1831016a0a418095356e236d55b07fe6

mdmower committed:
Unix file endings and remove execute permission

Change-Id: Iceaae3f7f20c3c6f23da807786b4cf4739c887c9

SevenMaxs committed:
Update & Fix ru.xml

Updating and minor fix of the translation for Russian localization.

Change-Id: Ic68e0580a12521e71a8da926901a30a31489a079

andi34 committed:
languages: fix and improve german translation

Change-Id: I7c5a0fbb1853836cfb5a7d1dfbbeb5f5613572b7

danparidae committed:
languages: add Polish translation

translated by @dadi11 & @linar10

Change-Id: I09a550c3ee99d44c4ada0bb7d71eb763471c7fda

sultanqasim committed:
patternpassword: update large grid password scheme

CM13 and up no longer use the two method scheme.

Change-Id: I21154becb5d7e77a54a5a07e649fe64829173a01

sultanqasim committed:
cryptfs: major overhaul and cleanup

- Don't upgrade HW encrypted Lollipop devices to Marshmallow crypto
- Fix support for passwords and patterns with an odd number of elements
- Remove unused code
- Fix build warnings

Change-Id: I25f015085e5c859d0353f42f6a2fbc7ccecd48ed

Dees-Troy committed:
Fix up logd make file rules

logd, the init.recovery.logd.rc, and other related files were
included by default if your device configs specified
TARGET_USES_LOGD := true which would be the case for any device
that has a full ROM tree instead of a minimal TWRP tree.

Change-Id: Ia1e55703631f0f22beab2f4d4479599b88539e1a

sndnvaps committed:
Fix RELINK of toybox for twrp

Change-Id: Ia609eec928278fc63e699d42799f74b4ada82221
Signed-off-by: sndnvaps <sndnvaps@gmail.com>

Dees-Troy committed:
Fix adopted storage when andsec flag is present

Change-Id: If903d838abc920f952f243482338188c15bb1dbf

Dees-Troy committed:
Restore some old decrypt functionality

Some of the convoluted convert_hex_ascii_to_key code is needed to
properly decrypt CM 12.1 patterns where grid size is larger than
3x3.

Change-Id: I497e17980046c60d2c69ba56e4b83c8b64b0b80e

magdeoz committed:
Update spanish translation

Change-Id: I4591a5fae396e7a993d4f70d80a1d1f3b2a83654

Dees-Troy committed:
Fix OEM compile

Change-Id: I39218f2485a2c2c95f0e400c31356bedc4973f6c

Dees-Troy committed:
Fix storage selection with adopted storage

Change-Id: Ifcba4e668579c3684df9e47e1c3faf5f4d1f3e83

Wechy77 committed:
TWRP: hungarian translation

add more strings
and fixes my mistakes in crowdin

ps2 remove whitespace

Change-Id: If79e61fba5c1b2e54be29286b23a1b418095b07c

that1 committed:
minuitwrp: retry opening fb0 if it failed

On some devices TWRP tries to access the framebuffer before all
device nodes have been created. Retry opening instead of crashing later.

Change-Id: I189a8fe80a8906b46fb6cece53c0bf83c00c0e09

mdmower committed:
partitions: Fix path based Find_Next_Storage

If a path is specified for Find_Next_Storage() and datamedia is
excluded, the function currently runs to the end of available
partitions and reports no other storage is available. Fix this by
restoring the original mount_point == search_path check to break the
loop. The subsequent partitions loop will skip data-media storage as
needed.

Change-Id: I92baed866cd36f13d851bd58ba5346e47edb142e

Dees-Troy committed:
Fix verifier_test build

Some people insist on running make without specifying any target
like recoveryimage or otapackage, which triggers a build of
verifier_test, so this fix will keep people from asking about
this build error anymore.

Change-Id: Ic955b7a3fb2599894e712fd8b418227da240d288

Dees-Troy committed:
Improve progress bar handling for backup / restore / image flash

The progress bar will now be updated during image backups, restores
and during image flashing (except for sparse images which will require
significant changes to libsparse, and except for mtd nand using
flash_utils).

The progress bar will now be updated mid-file for file systems (tar) so
the user will see changes even during large file backup / restore.

Add a new progress tracking class to simplify handling of progress bar
updates. The class will only update the progress bar 5 times a second to
reduce the CPU load from updating the GUI frequently which does affect
backup times.

Change-Id: Iff382faef3df1f86604af336c1a8ce8993cd12c5

Dees-Troy committed:
Remove busybox modprobe to fix slow performance for some devices

bionic process initialization calls personality
(specifically personality-8)
personality wants to load a kernel module
loading a kernel module calls /sbin/modprobe
loading /sbin/modprobe is a bionic process initialization
bionic process initialization calls personality
personality wants to load a kernel module
loading a kernel module calls /sbin/modprobe . . .

Before you know it, it takes 0.5 seconds to do anything.
Note: modprobe is still technically available, but the symlink
has been removed, so you can still call it directly by running
busybox modprobe if you like.

From what I can tell, this issue only affects 32 bit devices
with CONFIG_MODULES=y in the defconfig. The problem can be also
patched out of the kernel by commenting or otherwise removing
the block of code in kernel/exec_domain.c inside the CONFIG_MODULES
ifdef block

It is also possible to patch the problem in bionic libc by commenting
out or otherwise removing the __initialize_personality in bionic/libc/
bionic/libc_init_common.cpp file.

Change-Id: Iebac314616080ac18320d73b087980ac1b98b951

jcadduono committed:
Allow restoring to read-only file system

This allows the restoration of firmware or modems on read-only
mounted file systems. If the user is able to make a backup of
these partitions, it would only make sense they be allowed to
restore that backup. The current handling is dangerous in that
it happily wipes, but refuses to restore. :)

Note that it would be preferable to use the backup=emmc flag
instead in most cases, added by change:
https://gerrit.omnirom.org/#/c/17183

Change-Id: I32d47c8928dee61595c15a9db16d3c5b9a6d7183

jcadduono committed:
Don't add subpartitions to restore list

Their parents will take care of them.
Good parents.

Fixes double restoration of subpartitions when they are
also set to backup in the recovery fstab.

Change-Id: I876c179135e0cb00754e9a8cfc8eac164c4b7fd4

andi34 committed:
recovery: update italian translation

Translated by: Death Mask Salesman

Change-Id: I87f1f4417689b62c171264bfc05bf58e2ddfea03

jcadduono committed:
Merge code from upstream libtar + bug fixes

All updates and fixes applied from upstream libtar as of
March 1, 2016.

Debug flag is disabled, however non-debug output now
provides 1 line of useful output per object extracted.

I've also merged some fixes from CyanogenMod's
fork of libtar:

From: Tom Marshall <tdm@cyngn.com>
Date: Thu, 11 Feb 2016 16:24:40 -0800
Subject: libtar: Cleanup, secure, and extend numeric fields
Commit: e18b457ea1cbf6be1adc3b75450ed1c737cd82ea

From: Tom Marshall <tdm@cyngn.com>
Date: Thu, 11 Feb 2016 12:49:30 -0800
Subject: libtar: Make file sizes 64-bit clean
Commit: e628c2025549a24018bc568351465130a05daafb

From: Tom Marshall <tdm@cyngn.com>
Date: Thu, 17 Apr 2014 09:39:25 -0700
Subject: libtar: Add methods for in-memory files
Commit: 8ec5627a8ff0a91724c6d5b344f0e887da922527

From: Tom Marshall <tdm@cyngn.com>
Date: Wed, 2 Jul 2014 09:34:40 -0700
Subject: libtar: Fix hardlink extract
Commit: 166d83a51e0c51abcea37694dbd7df92d03c1f56

From: philz-cwm6 <phytowardt@gmail.com>
Date: Sat, 26 Apr 2014 01:11:35 +0200
Subject: libtar: Various bug fixes and enhancements
Commit: a271d763e94235ccee9ecaabdb52bf4b9b2f8c06
(Some of this was not merged in, as better solutions were
available from upstream libtar)

From: Tom Marshall <tdm@cyngn.com>
Date: Wed, 9 Apr 2014 09:35:54 -0700
Subject: libtar: Add const qualifiers to reduce compile warnings
Commit: 0600afa19fe827d06d3fcf24a7aabd52dbf487b4

Change-Id: I6d008cb6fdf950f835bbed63aeb8727cc5c86083

bigbiff committed:
TWRP Install Reboot Option

This is a simple patch to allow users to reboot
after installation of zip queue. Current timeout
is 5 seconds before reboot.

Change-Id: I1a18cbf7025086e35031ab7241b330babebb8343

daveyannihilation committed:
Fix horrible white UI to be in line with new TWRP 3.0 theme

jollaman999 committed:
gui/themes_multirom/1080x1920: Get images from z31s1g's Play theme

andi34 committed:
recovery: czech translation

Uploaded on XDA from XDA-User "kodein"
http://forum.xda-developers.com/showpost.php?p=65371227&postcount=1400

Change-Id: I6c3c4fe90b91a21400b11541f3895d92cc80ea1d

that1 committed:
gui: rewrite theme loading code

- Reduce code duplication for parsing of include files
- Fix crash when styles and templates from included files are referenced
- Fix bug with path concatenation if more than one file is included

Change-Id: I06cd720188df272180fb359097afd73a85952fa8

Myself5 committed:
use 1080x1920 theme for 720x1280 devices too

The Sony Xperia Z3 Compact for example is using a 720x1280 screen, and can use the 1080x1920 theme just fine.

that1 committed:
gui: fix font scaling for zip themes

All fonts were extracted to /tmp/extract.bin which was deleted after
the initial load, so reloading for scaling failed.

- extract fonts to /tmp with original name and don't delete them
- minor code cleanup

Change-Id: If8a0f657a7ef4c418fd5cc8550a24de44a38f303

Dees-Troy committed:
DataManager Updates

The goal of this change is to make DataManager use InfoManager to reduce
code duplication.

Change-Id: Ia4f4c4324453a192995e0f442db0a03628c13e46

Dees-Troy committed:
Improve input box text handling

Change-Id: I79cfaac1adf85d1ee5514bf5266930823c1a15ff

Dees-Troy committed:
Change to dd for wiping crypto partitions

flash_image was spitting an error. We are also specifying a size
so that we do not risk writing beyond the partition.

Change-Id: Ie3a0d77db96e54357238595934c3df0e3b680823

Dees-Troy committed:
Finally update NOTICE file to contain GPL v3

Individual files have contained the GPLv3 notice, but now the main
NOTICE file will indicate GPLv3 to better reflect the license of
the project.

Change-Id: I277b88e5853b7a7fc64f2305a5d4a3fff4e88e7d

Dees-Troy committed:
Fix year in NOTICE

Change-Id: Iaade6424f23855b000bd282d3d00fba5d061c535

Dees-Troy committed:
Detect and install custom themes as part of zip installs

Change-Id: I7ec7133e53fca92ebe49a9a41161754a9138141e

jcadduono committed:
Flashutils: detect /dev/block/sd* as emmc

Change-Id: I067a58e5053748b9613ea6b4ac76a1c5285abc7d

Dees-Troy committed:
Add language_helper.py

This script is intended to make managing the language files a
little easier.

This script reads the English and supplied other language files and
compares the 2 and reorders and rewrites the other language to a new
XML file such that all the strings are placed in the same order as
the English file. It will place commented out string lines for items
that are not present in the new file and will not include any strings
in the new file that are no longer present in the English source.
There is also a version tag that may be compared if present between
the English and other language in case a translation string changes.

Change-Id: If49d7efbf99c033d9736af971a78ed8455ac5c6f

Dees-Troy committed:
Stop trying to load /script/ui.xml

I think this feature was used way back in the very early days of
TWRP 2.0 on the Kindle Fire and has not been used since. I see no
reason to keep it around for now and we can always reinstate it
later if for some reason we need this feature again.

Change-Id: Ie9f3e5db7190732f813ae1e152aab2d85a4ca98c

Dees-Troy committed:
Stop loading languages for the splash screen

We don't translate anything for the splash screen currently so
this will slightly reduce startup times and prevent the code from
running when it's not needed.

Change-Id: I8a8e8be5247d8e8b148db578c12bce5d370123a8

mdmower committed:
progress: Initialize display percentage variable

progress_percent can be set to an uninitilized display_percent if
total_backup_size == 0.

Change-Id: I704bf9a1250e95df65ef8790036f94c0a77da8d8

danparidae committed:
languages: add Reboot Option translation

Change-Id: I33acec5b4d0a371978bf2a6ab3beb26d0d18400f

Dees-Troy committed:
HACK: fix input focus on terminal

This makes the terminal grab focus no matter what on SetPageFocus
which isn't ideal but it's unlikely that any other visible input
elements will be on the page with a terminal anyway.

Credit _that for pointing me in the right direction for this fix.

Change-Id: I2b0e6fc1169521f58e863c58f7ef3578413dfb21

z31s1g committed:
language: add 2 new strings to DE

Change-Id: Ic9d51ddb7a8e47c62399d2e3101e788994276104

Dees-Troy committed:
3.0.1-0

Change-Id: Iecf5c3f70559ef90e18f4807f80707fa5a3c78f0

jollaman999 committed:
multirom: Fix build error

SlimShadys committed:
language: add 2 new strings in it.xml

Change-Id: I45a6a75d2b9332e523a7f61f19594c4029665ed7

SevenMaxs committed:
languages: add 2 new strings for RU

Added translation Reboot Option and minor fix for Russian localization.

Change-Id: Ibeade0ce85ac65daf68dfd6b085a32148f66d1cb

Dees-Troy committed:
Really remove all old SP* flags and code

Change-Id: I5ece02a8721e1b7541fb5182366a55fd0f037a7a

jollaman999 committed:
Fix duplicated main screen after decrypted

jollaman999 committed:
themes_multirom: Add Reboot Option

Commit:
fix typo

Change-Id: I65f16530565c4b510274c72227a5f1483519a677

Commit:
Greek Translation from the updated en.xml

Change-Id: I5cd1b9c394c2777503e5567aaaa01210cfdb4caa

sultanqasim committed:
gui: don't mix up display value with actual value

In the case of password entry, the two differ. Password entry was broken
by the earlier commit entitled "Improve input box text handling" because
it started using the displayValue as the actual value, even in cases where
they differ because of a mask value.

Change-Id: Iaf5a67e1d928f34595962a1f1b80eebb64e8b493

Dees-Troy committed:
3.0.2-0

Change-Id: I28fa04cb90dd4255db9dc5d771416ab8d456de46

Commit:
translate install reboot option to dutch

Change-Id: I23b2c44afca5c903becd201503f81766828c9304

Wechy77 committed:
Update hungarian translation

Change-Id: Iabd66596c0d1b83935c4adc4ac048f3881284269

muhammadfahadbaig committed:
Black screen patch for some HiSilicon devices (FBIOPAN_DISPLAY)

Change-Id: Ib114dcc7b48219142602a2fbc256e2a0258b26e0

supergordon committed:
Partition: Fix parsing of several twrp fstab flags

fix parsing when using storagename=, backupname= and userdataencryptbackup= in twrp.fstab
also split encryptable= and forceencrypt= into seperate blocks
and make storage=Y working

Change-Id: I674f7ffe7f9e4ff121c4dccf79096360e044535c

mdmower committed:
Rewrite TWRP fstab flag processing (Fixed for MultiROM TWRP)

* Do not flip/flop between std::string and char* when no std::string
  specific functions are applied.
* Remove the need to manually count flag string lengths
* Move checks for Display_Name, Storage_Name, and Backup_Display_Name
  out of flag processing

Change-Id: I24d432c222124012b2a98d27598e42d0944f7da2

AndroPlus-org committed:
Languages: Update Japanese translation

Change-Id: I4ec685a9d70173c1e008beb61cdead6af5c2aae0

AndroPlus-org committed:
Languages: Add font for Japanese

Japanese font is NotoSans which is licensed under SIL OFL.

Change-Id: I6f50edbca72c7e58ce0e96aee98a0303bd1a4254

Commit:
Improve greek translation

Some words are too big for the boxes

Change-Id: I48c2361dd9c7937361c3fb63b9bd8834c57fe0e0

Dees-Troy committed:
Fix decrypt of odd number length PIN on hardware crypto

I am not sure if we are really fixing anything other than we are
allowing the decrypt process to continue. On hardware crypto the
password never seems to match what is expected from the data in
the footer, probably because the data is not stored in the footer
and TZ does all the work. Still, if it works, it is hard to fault
the patch.

Change-Id: Ibbb286382e82523bec2064f51fa07194f84820c2

mdmower committed:
Rewrite FS fstab flag processing

* If 'ro' fsflag detected, set Mount_Read_Only = true
* Only output human readable Mount_Options to log
* Match fsflags identically (not just first n chars) since there are
  no fs_flags which take arguments
* Match new processing method introduced in TW fstab flag processing

Change-Id: Iefdb76016be90a131b0d627d0cd3f18d2eb1a008

mdmower committed:
Fixup reboot after install variables

* Create correctly named data manager variable 'tw_install_reboot'
* Make configurable through Settings
* Remove closing tag space in theme files (not personal preference,
  but should match surrounding code style)
* Fix the "two presses needed to enable" issue for the checkbox on the
  install screen

Change-Id: I0ceb60b4888a672e2f0ecb5672b03cc86615621a

mdmower committed:
Finish creating disable_free_space_check variable

* Create data manager variable for 'tw_disable_free_space'
* Make configurable through Settings
* Fix the "two presses needed to enable" issue for the checkbox on the
  backup screen
* No need to capitalize every word, but do specify precedes backup

Change-Id: Id436cef13e4ca9349618420aac03862ec4e3c35e

Commit:
Restore TW_SCREEN_BLANK_ON_BOOT functionality.

This is required for a number of devices, particularly now that custom
graphics support is deprecated.

This change is required to officially support TWRP on the Huawei Plank
(aka the Honor 7).

Change-Id: Ie16ec557c75654c9c9e7a2877aee3339fc214d05

mdmower committed:
Remove char/string conversions in fstab processing

There a few char->string conversions between functions which handle
different parts of fstab line processing, but there are no uses of
string functions. Pass char arrays/pointers around instead.

Change-Id: I976a9a54ee8dcfb6194cadcac6a34e467602003b

mdmower committed:
Wait to mount until after fstab processed

The necessity to process fstab twice stems from mounting partitions
while still processing. Instead, wait to finish setup of /data, /cache
and storage parameters until after fstab has been processed (once).

Change-Id: Id77e1edbab5eb68a7cd4a1f34953d819a043d47a

jollaman999 committed:
partition.cpp: Fix build error for MultiROM TWRP

CaptainThrowback committed:
crypto: remove redundant convert_key_to_hex_ascii call

- Breaks decryption on some hw_crypto devices
- Default value already defined in preceding ifdef
- PS2: Move crypt_params definition prior to ifdef
  (matches corresponding code from CAF)

Huge thanks to @beaups for figuring out the issue!

Change-Id: I1fd4e3a4862f022b17a555773feb1d6deac9d34c

jollaman999 committed:
Revert "partition.cpp: Fix build error for MultiROM TWRP"

This reverts commit 78430567baad143423cf22d2ae3975e1b54da7aa.

jollaman999 committed:
Revert "Wait to mount until after fstab processed"

This reverts commit ed521c9d8c28122b7d436b326d3e6b96c998832a.

jollaman999 committed:
Revert "Remove char/string conversions in fstab processing"

This reverts commit 67a093d017e1bc30e2c666a903579c33bbd23a35.

jollaman999 committed:
Revert "Rewrite FS fstab flag processing"

This reverts commit a0b2c06a6c1e386c38b8d495504187478d3dcfdf.

jollaman999 committed:
Revert "Rewrite TWRP fstab flag processing (Fixed for MultiROM TWRP)"

This reverts commit 55896e596230986135cc9de5a0845c8ee4f474be.

that1 committed:
gui: fix Enter key in terminal

Translating CR to LF was an old hack that breaks Enter in nano.

Change-Id: If259e8b133f2f56f4126a1fb68856bf014a2daae

mdmower committed:
GUI: Use standard routine for auto-reboot

If 'reboot after install' is enabled, follow the standard reboot
routine to process 'No OS' and 'Install SuperSU' checks.

Since the system reboot routine is repeated several times, it has been
separated into its own routine.

Change-Id: I4c1882ecb8dc2d153ec2378b2e0973ab1a2a8278

webgeek1234 committed:
Move force_rgb_565 up.

Changes to fb_var_screeninfo can cause changes to
fb_fix_screeninfo. So, write back the force changes before fi is
read.

Change-Id: I721a960fa401ac5203ffc90bd3bfa2d7d0cfb292

CaptainThrowback committed:
Add typecode to sgdisk for proper detection in Android

- Fixes the "Unsupported SD Card" error when partitioning w/TWRP

PS2:
Use hex code instead of GUID to make code more succinct
Update typecode for Linux swap to be consistent

PS3: Revert PS2 changes

Change-Id: I5aec780aa4dfb7bd9025ae8cbceda13bfaaa0b03

npjohnson commented:
@jollaman999 Can you add the most recent Theme fixes from my repo to this pull request?

We have all resolutions accurately themed.

Commit:
wait for cryptfs device node to come up

this is to prevent a race condition to occur when mounting an adopted
storage volume just after it was decrypted. After setting up the
decrypted volume using cryptfs_setup_ext_volume(), Decrypt_Adopted()
immediately calls Mount(), which will call Check_FS_Type(), which will
fail in blkid_new_probe_from_filename, as the device node was not
created yet.

Change-Id: Ic5a274c2066ab278c9b7f1c8b83b820e552ca344

mdmower committed:
Move cursor after appending date to backup name

* Send the cursor to the end of the input text

Change-Id: Id4f20717134cd38188cc1107f73dcd6bb995f377

mdmower committed:
Remove unused identifier from NotifyTouch

The 'state' parameter in GUIAction::NotifyTouch() is used in the
function.

Change-Id: I0744c7aa1542d771dbd2916638fbdd53e9681f86

mdmower committed:
Remove curtainClose() remnants

Change-Id: I6ae96b60d34fa4e4b23cd1e5a4fffff3d00451c7

Dees-Troy committed:
Allow running of a custom script after factory reset

Change-Id: I33e6d72bf93103052551f33da51ba4af9a5f88c9

Dees-Troy committed:
Add build flag for forcing single buffer mode

RECOVERY_GRAPHICS_FORCE_SINGLE_BUFFER := true

Change-Id: Id5144ea772c3b7ae382b064c41c23acdd0decb84

maxwen committed:
recovery: allow usage of TARGET_CRYPTFS_HW_PATH

only if not defined fallback to the default
more flexible device config where the cryptfs_hw is located

Change-Id: I7d1c18eeae877e48dceff06a7cfead28c89797b4

that1 committed:
adbbu: fix include path, now builds again in Omni 4.4

Change-Id: Idee80db34a0685abb4bec50383dcbc4911c1a341

bigbiff committed:
Fix md5 creation from adb backup patch.

Change-Id: Id90da89c4eb50ed695a88f29859d1276ed8d032b

Dees-Troy committed:
Make input whitelist and blacklist work for mouse devices too

Includes some minor code clean up. Also we are now outputting the
name of the first mouse device that we encounter to make it easier
to identify which device(s) may need to be blacklisted.

Change-Id: I515baf92967390edd224728f3a7092239138e6b8

that1 committed:
Add git revision to startup message in recovery.log

This should make it easier to find the exact source for
a specific version of the binary.

Change-Id: I7e04dd9a3205cc643150a588b69b4bd691212ecb

jollaman999 committed:
Optimize and reduce size of PNG files

Done using:
find -type f -name *.png -exec optipng -o7 -nc -fix {} +
(-nc is required as TWRP png lib can't handle color type changes)
find -type f -name *.png -exec advdef -4 -z {} +
(use Zopfli compressor on insane prefix to reduce size even more)

jollaman999 committed:
multirom: Fix for Restore_Partition changes

jollaman999 committed:
Revert "multirom: Fix for Restore_Partition changes"

This reverts commit 5b819bce585de1e274d68050ac5ab05a516e6ea9.

jollaman999 committed:
Revert "Fix md5 creation from adb backup patch."

This reverts commit 821ac88c02e5e9e4065a55032430a44198a8c035.

jollaman999 committed:
Revert "adbbu: fix include path, now builds again in Omni 4.4"

This reverts commit 5fbb6f2ea362f66be753619cffc63faedcadd0e0.

jollaman999 committed:
Revert "ADB: Add adb backup for TWRP."

This reverts commit e5eabff1657001b05f8f07e89f9aff7c03b9b84b.

nailyk-fr committed:
minui: Fix gr_set_font() build issue on cm-13.0 tree.

- Add a fake function which fill the vendor/cm/charger/healthd_board_cm.cpp requirements.

Solve omnirom/android_bootable_recovery#84 issue.

Change-Id: Ia9bf898006da7e4f4ee6a942584fe2dd3c88f8ba

nailyk-fr committed:
Gui: replace pow() function

The old pow function doesn't work for other power than 2
- replace pow function by a squared one

Solve the TeamWin/Team-Win-Recovery-Project#646 issue.

Change-Id: Id177300d45a7b49ff983795288434d910eb35c2a

jollaman999 committed:
Fix merge error
